### PR TITLE
Integrated beforeIndexRemoved onIndexModule extension point

### PR DESCRIPTION
**Pull request by owaiskazi19**
Signed-off-by: Owais Kazi <owaiskazi19@gmail.com>

- [x] Integrated beforeIndexRemoved sub method of onIndexModule

### CODE REVIEW

**Review by saratvemulapalli — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
@owaiskazi19 could you rebase with main and it should fix the checks.

File: src/main/java/transportservice/RunPlugin.java [hunk 1]
```diff
@@ -25,9 +25,13 @@
...
+import org.opensearch.index.IndicesModuleNameResponse;
+import org.opensearch.index.IndicesModuleRequest;
+import org.opensearch.index.IndicesModuleResponse;
 import org.opensearch.indices.IndicesModule;
 import org.opensearch.indices.breaker.CircuitBreakerService;
 import org.opensearch.indices.breaker.NoneCircuitBreakerService;
+import org.opensearch.plugins.PluginsOrchestrator;
 import org.opensearch.search.SearchModule;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.*;
```
Comment (saratvemulapalli) [inline src/main/java/transportservice/RunPlugin.java]: nit: Why do we have a `*` import. 
Can we add a rule to spotless to avoid this?

Comment (owaiskazi19) [reply src/main/java/transportservice/RunPlugin.java]: Yeah that's a good idea. I have removed the `*` though. I'm not aware of such rule but will take a look.

**Review by joshpalis — APPROVED** [review]
looks great

**Pull request merged**